.github/workflows/lint-tidy.yml

```diff
@@ -0,0 +1,59 @@
+name: Lint & Tidy
+on:
+  workflow_call:
+  workflow_dispatch:
+
+env:
+  RUST_BACKTRACE: 1
+  SHELL: /bin/bash
+  SCCACHE_GHA_ENABLED: "true"
+  RUSTC_WRAPPER: "sccache"
+  CCACHE: "sccache"
+  CARGO_INCREMENTAL: 0
+
+jobs:
+  lint-tidy:
+    name: Lint & Tidy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        if: github.event_name != 'pull_request_target'
+        with:
+          fetch-depth: 2 # This is necessary for `test-tidy`.
+      # This is necessary to checkout the pull request if this run was triggered via a
+      # `pull_request_target` event.
+      - uses: actions/checkout@v4
+        if: github.event_name == 'pull_request_target'
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 2 # This is necessary for `test-tidy`.
+      - name: Run sccache-cache
+        uses: mozilla-actions/sccache-action@v0.0.4
+      - name: Set LIBCLANG_PATH env # needed for bindgen in mozangle
+        run: echo "LIBCLANG_PATH=/usr/lib/llvm-14/lib" >> $GITHUB_ENV
```

Comment:
I would also specify `CLANG_PATH` to point to clang-14 . (Otherwise there is chance we run into https://github.com/rust-lang/rust-bindgen/issues/2682 when upgrading the ubuntu image to 24.04 and the default clang version changes. Explicitly setting CLANG_PATH, should make it clearer that both versions need to be in sync.

Reply:
There is no clang 14 in 24.04. I usually just hope for the best (if it's not broken, don't fix it), but IMO clang-sys and bindgen should fix their stuff.